### ISSUE TITLE
Determine the version of Kafka client lib to use

### ISSUE THREAD

**Issue by fhopf** [logstash-plugins/logstash-input-kafka#188]
This is an idea by @jordansissel expressed on Twitter. It would be really nice if the plugin could guess the version of Kafka it connects to and use the correct client lib. Not sure if this is possible at all but would be very convenient for the user.

**Comment by original-brownbear** [comment]
@fhopf @jordansissel I wonder if this has become outdated?

The more recent versions of this plugin (https://github.com/logstash-plugins/logstash-input-kafka/blob/master/CHANGELOG.md#630) now use Kafka `10.2.x` which is backwards-compatible with all `10.x`. (see https://kafka.apache.org/0102/documentation.html#upgrade_1020_notable).

So this would only help `0.9.x` clusters and older. Since the APIs on the Java client we have to call are not 100% the same across versions, this would be a lot of effort just to support the older Kafka versions?

**Comment by jordansissel** [comment]
@original-brownbear I don't think it's outdated, not yet anyway. The claim of backwards compatibility is nice, but a many-years history of incompatibility makes me cautious - Trust, but verify. Even if we ship Kafka 10.2.x clients, we'll still need to fully test these clients against Kafka 10.0 and 10.1 anyway in order to provide confidence in a release. (Just like we do with the Logstash Elasticsearch output which we test against many many versions of Elasticsearch)

So separating things out:

* No, we don't have to do this kind of detection, but it might be nice for users.
* Yes, Kafka claims they are investing in backwards compatibility.
* Yes, we will still need to verify this claim every time we upgrade the Kafka client and before releasing an updated plugin.

**Comment by original-brownbear** [comment]
@jordansissel agreed we should test :) (https://github.com/elastic/logstash/issues/7100)
